source/server/admin/stats_request.h

```diff
@@ -87,31 +108,42 @@ class StatsRequest : public Admin::Request {
   // current phase.
   void populateStatsForCurrentPhase(const ScopeVec& scope_vec);
 
-  // Populates all the metrics of the templatized type from scope_vec. Here we
-  // exploit that Scope::iterate is a generic templatized function to avoid code
-  // duplication.
-  template <class StatType> void populateStatsFromScopes(const ScopeVec& scope_vec);
+  virtual Stats::IterateFn<Stats::TextReadout> saveMatchingStatForTextReadout() PURE;
+  virtual Stats::IterateFn<Stats::Gauge> saveMatchingStatForGauge() PURE;
+  virtual Stats::IterateFn<Stats::Counter> saveMatchingStatForCounter() PURE;
+  virtual Stats::IterateFn<Stats::Histogram> saveMatchingStatForHistogram() PURE;
 
-  // Renders the templatized type, exploiting the fact that Render::generate is
-  // generic to avoid code duplication.
-  template <class SharedStatType>
-  void renderStat(const std::string& name, Buffer::Instance& response, StatOrScopes& variant);
+  virtual void processTextReadout(const std::string& name, Buffer::Instance& response,
+                                  const StatOrScopes& variant) PURE;
+  virtual void processGauge(const std::string& name, Buffer::Instance& response,
+                            const StatOrScopes& variant) PURE;
+  virtual void processCounter(const std::string& name, Buffer::Instance& response,
+                              const StatOrScopes& variant) PURE;
+  virtual void processHistogram(const std::string& name, Buffer::Instance& response,
+                                const StatOrScopes& variant) PURE;
 
   // Sets the chunk size.
   void setChunkSize(uint64_t chunk_size) { chunk_size_ = chunk_size; }
 
-private:
+protected:
```

Comment:
Hmm. WDYT of leaving this private and moving some of these to the subclass (e.g. render_ in particular), or accessing them from subclasses via protected API?